exercises/practice/robot-simulator/.meta/example.php

```diff
@@ -2,68 +2,68 @@
 
 declare(strict_types=1);
 
-class Robot
+class RobotSimulator
 {
-    public const DIRECTION_NORTH = 'north';
-    public const DIRECTION_EAST = 'east';
-    public const DIRECTION_SOUTH = 'south';
-    public const DIRECTION_WEST = 'west';
+    private const DIRECTION_NORTH = 'north';
+    private const DIRECTION_EAST = 'east';
+    private const DIRECTION_SOUTH = 'south';
+    private const DIRECTION_WEST = 'west';
+
+    /** @param int[] $position */
+    public function __construct(
+        private array $position,
+        private string $direction,
+    ) {
+    }
 
     /**
-     *
-     * @var int[]
+     * Move the Robot according to instructions: R = Turn Right, L = Turn Left and A = Advance
      */
-    protected $position;
+    public function instructions(string $instructions): void
+    {
+        if (!preg_match('/^[LAR]+$/', $instructions)) {
+            throw new InvalidArgumentException('Malformed instructions');
+        }
 
-    /**
-     *
-     * @var string
-     */
-    protected $direction;
+        foreach ($this->mapInstructionsToActions($instructions) as $action) {
+            $this->$action();
+        }
```

Comment:
(non-blocking) This keeps the idea of the previous example. But I feel like having dynamic function call is more complex than needed.

I suggest to inline the mapping with the function call:

```suggestion
        foreach (str_split($instructions) as $instruction) {
            match ($instruction) {
              'A' => $this->advance(),
              'L' => $this->turnLeft(),
              'R' => $this->turnRight(),
            }
        }
```

Reply:
Thanks for requesting this. Also used `match()` in `advance()`, as it is much more concise and understandable.